### PR TITLE
BFB porting ice_cldliq_wet_growth to scream cpp

### PR DESCRIPTION
**Pull request by xyuan**
BFB porting ice_cldliq_wet_growth to scream cpp

BFB

### CODE REVIEW

**Comment by xyuan** [comment]
Hope this time, it will work. The test results are not reproductible on summit while I cam able to build and run without any issues.

sizeof(Real) 8 avx  packsize 1 compiler GCC FPE off #threads 1
Comparing with test8 at tol 0.0e+00
--- checking case # 1, it = 1/1 ---
--- checking case # 2, it = 1/6 ---
--- checking case # 2, it = 2/6 ---
--- checking case # 2, it = 3/6 ---
--- checking case # 2, it = 4/6 ---
--- checking case # 2, it = 5/6 ---
--- checking case # 2, it = 6/6 ---
--- checking case # 3, it = 1/1 ---
--- checking case # 4, it = 1/6 ---
--- checking case # 4, it = 2/6 ---
--- checking case # 4, it = 3/6 ---
--- checking case # 4, it = 4/6 ---
--- checking case # 4, it = 5/6 ---
--- checking case # 4, it = 6/6 ---

File: components/scream/src/physics/p3/p3_functions_ice_cldliq_wet_growth_impl.hpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+void Functions<S,D>
+::ice_cldliq_wet_growth(const Spack& rho, const Spack& temp, const Spack& pres, const Spack& rhofaci, const Spack& f1pr05,
+                        const Spack& f1pr14, const Spack& xxlv, const Spack& xlf, const Spack& dv,
+                        const Spack& kap, const Spack& mu, const Spack& sc, const Spack& qv, const Spack& qc_incld, 
+                        const Spack& qitot_incld, const Spack& nitot_incld, const Spack& qr_incld, 
+                        Smask& log_wetgrowth, Spack& qrcol, Spack& qccol, Spack& qwgrth, Spack& nrshdr, Spack& qcshd)
+{
+   constexpr Scalar qsmall = C::QSMALL;
+   constexpr Scalar tmelt  = C::Tmelt;
+   constexpr Scalar twopi  = C::Pi*sp(2.0);
```
Comment (jgfouca) [inline components/scream/src/physics/p3/p3_functions_ice_cldliq_wet_growth_impl.hpp]: Please use integer literals where possible, so you don't need to worry about overpromotion.
```
constexpr Scalar twopi  = C::Pi*2;
```

File: components/scream/src/physics/p3/p3_functions_f90.hpp [hunk 2]
```diff
@@ -843,6 +843,29 @@ void ice_nucleation_f(Real temp, Real inv_rho, Real nitot, Real naai,
...
+struct IceWetGrowthData
+{
+  // Inputs
+  Real rho, temp, pres, rhofaci, f1pr05, f1pr14, xxlv, xlf, dv, kap, mu, sc, qv, qc_incld;
+  Real qitot_incld, nitot_incld, qr_incld;
+
+  // In/Outs
+  bool log_wetgrowth;
+  
+  // In/Outs
```
Comment (jgfouca) [inline components/scream/src/physics/p3/p3_functions_f90.hpp]: You don't need to mark the in/outs region twice.

File: components/scream/src/physics/p3/p3_functions_ice_cldliq_wet_growth_impl.hpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+   const auto qitot_incld_ge_small = qitot_incld >= qsmall;
+   const auto qc_qr_incld_ge_small = (qc_incld + qr_incld) >= sp(1.0e-6);
+   const auto qccol_qrcol_ge_small = (qccol + qrcol) >= sp(1.0e-10);
+
+   const auto any_if     = qitot_incld_ge_small && qc_qr_incld_ge_small && t_is_negative;
+   const auto any_if_col = any_if && qccol_qrcol_ge_small;
+
+   const Spack zerodeg{tmelt};
+   const Spack e0 = polysvp1(zerodeg, zero);
+   Spack qsat0{0.0};
```
Comment (jgfouca) [inline components/scream/src/physics/p3/p3_functions_ice_cldliq_wet_growth_impl.hpp]: These a can all be `0`.

File: components/scream/src/physics/p3/tests/p3_ice_cldliq_wet_growth_unit_tests.cpp [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    KTH::view_1d<IceWetGrowthData> self_host("self_host", Spack::n);
+    view_1d<IceWetGrowthData> self_device("self_host", Spack::n);
+    std::copy(&self[0], &self[0] + Spack::n, self_host.data());
+    Kokkos::deep_copy(self_device, self_host);
+
+
+    // Get data from fortran
+    for (Int i = 0; i < Spack::n; ++i) {
+      ice_cldliq_wet_growth(self[i]);
+     }
```
Comment (jgfouca) [inline components/scream/src/physics/p3/tests/p3_ice_cldliq_wet_growth_unit_tests.cpp]: Bad indent (one space too many)

File: components/scream/src/physics/p3/tests/p3_ice_cldliq_wet_growth_unit_tests.cpp [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+      {1.252000E+01, 3.145000E+03, 8.213000E+04, 9.290000E-03, 1.034000E-02, 1.069000E+00, 0.123000E+00, 1.221000E-02, 1.952000E-06, 6.596000E-05, 2.340006E-01, 5.632000E-01, 8.452000E-00, 1.003000E-01, 2.145000E-00, 1.734000E+04, 9.212000E-04, false, 6.732276E-05, 7.321873E-03, 8.234000E-00, 9.023000E-01, 7.201000E-00},
+      {1.352000E+01, 4.742000E+03, 1.014000E+05, 1.234000E-02, 1.456000E-02, 1.069000E+00, 0.123000E+00, 1.221000E-02, 9.952000E-07, 6.596000E-05, 4.123000E-01, 6.128000E-01, 9.076000E-00, 2.831000E-01, 3.902000E-00, 1.734000E+04, 1.023000E-03, false, 7.902887E-05, 9.032908E-03, 9.021000E-00, 1.092000E-01, 8.096000E-00}     
+    };
+    
+    // Sync to device
+    KTH::view_1d<IceWetGrowthData> self_host("self_host", Spack::n);
+    view_1d<IceWetGrowthData> self_device("self_host", Spack::n);
+    std::copy(&self[0], &self[0] + Spack::n, self_host.data());
+    Kokkos::deep_copy(self_device, self_host);
+
```
Comment (jgfouca) [inline components/scream/src/physics/p3/tests/p3_ice_cldliq_wet_growth_unit_tests.cpp]: Only a single blank line between sections.

File: components/scream/src/physics/p3/tests/p3_ice_cldliq_wet_growth_unit_tests.cpp [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+    // Get data from fortran
+    for (Int i = 0; i < Spack::n; ++i) {
+      ice_cldliq_wet_growth(self[i]);
+     }
+
+
```
Comment (jgfouca) [inline components/scream/src/physics/p3/tests/p3_ice_cldliq_wet_growth_unit_tests.cpp]: Only a single blank line between sections.

File: components/scream/src/physics/p3/tests/p3_ice_cldliq_wet_growth_unit_tests.cpp [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+    Kokkos::parallel_for(RangePolicy(0, 1), KOKKOS_LAMBDA(const Int& i) {
+     // Init pack inputs
+     Spack rho,temp, pres,rhofaci,f1pr05,f1pr14,xxlv,xlf,dv,kap,mu,sc, 
+           qv,qc_incld,qitot_incld,nitot_incld,qr_incld;
+
+     Smask log_wetgrowth;
+
+     Spack qrcol,qccol,qwgrth,nrshdr,qcshd;
+
+      for (Int s = 0; s < Spack::n; ++s) {
```
Comment (jgfouca) [inline components/scream/src/physics/p3/tests/p3_ice_cldliq_wet_growth_unit_tests.cpp]: bad indent

File: components/scream/src/physics/p3/tests/p3_ice_cldliq_wet_growth_unit_tests.cpp [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+      REQUIRE(self[s].log_wetgrowth == self_host(s).log_wetgrowth);
+      REQUIRE(self[s].qrcol         == self_host(s).qrcol);
+      REQUIRE(self[s].qccol         == self_host(s).qccol);
+      REQUIRE(self[s].qwgrth        == self_host(s).qwgrth);
+      REQUIRE(self[s].nrshdr        == self_host(s).nrshdr);
+      REQUIRE(self[s].qcshd         == self_host(s).qcshd);
+    }
+  }
+
+
```
Comment (jgfouca) [inline components/scream/src/physics/p3/tests/p3_ice_cldliq_wet_growth_unit_tests.cpp]: Only a single blank line between sections.

**Comment by jgfouca** [comment]
@xyuan , is this still a WIP?

**Comment by xyuan** [comment]
> @xyuan , is this still a WIP?

nope, I put WIP because the jenkins test is stucked

**Pull request merged**